nemo/core/classes/common.py

```diff
@@ -456,6 +458,17 @@ def from_config_dict(cls, config: 'DictConfig', trainer: Optional['Trainer'] = N
                         imported_cls = cls
                     accepts_trainer = Serialization._inspect_signature_for_trainer(imported_cls)
                     if accepts_trainer:
+                        if trainer is None:
```

Comment:
Revert this change - this is a hack that is dangerous. If the model cannot load without trainer, detect it in its constructor and hard crash instead of hacks like this.

As of now, this will introduce silent bugs - the model will load ofc, but then the user may setup another trainer with multiple gpus or other config and will forget to use model.set_trainer() to reflect that change. 

More so - the dataloaders and other things may refer to wrong trainer for ASR and TTS. 

Also, this is a core change, who else reviewed this?

Reply:
@ramanathan831 @borisfom

Reply:
@borisfom can comment further, I had similar concerns/thoughts, hence suggested having the trainer in the scripts/export.py

Reply:
reverted these changes here: https://github.com/NVIDIA/NeMo/pull/4039